autogalaxy/plot/mat_wrap/get_visuals.py

```diff
@@ -0,0 +1,451 @@
+from typing import List, Union
+
+import autoarray as aa
+
+from autoarray.plot.mat_wrap import get_visuals as gv
+
+from autogalaxy.plot.mat_wrap.include import Include1D
+from autogalaxy.plot.mat_wrap.include import Include2D
+from autogalaxy.plot.mat_wrap.visuals import Visuals1D
+from autogalaxy.plot.mat_wrap.visuals import Visuals2D
+
+from autogalaxy.util import error_util
+
+from autoarray.structures.grids.two_d.grid_2d import Grid2D
+from autoarray.structures.grids.two_d.grid_2d_irregular import Grid2DIrregular
+
+from autogalaxy.imaging.fit_imaging import FitImaging
+from autogalaxy.profiles.light_profiles.light_profiles import LightProfile
+from autogalaxy.profiles.mass_profiles.mass_profiles import MassProfile
+from autogalaxy.galaxy.galaxy import Galaxy
+from autogalaxy.plane.plane import Plane
+
+
+class GetVisuals1D(gv.GetVisuals1D):
+    def __init__(self, include: Include1D, visuals: Visuals1D):
+        """
+        Class which gets 1D attributes and adds them to a `Visuals1D` objects, such that they are plotted on 1D figures.
+
+        For a visual to be extracted and added for plotting, it must have a `True` value in its corresponding entry in
+        the `Include1D` object. If this entry is `False`, the `GetVisuals1D.get` method returns a None and the attribute
+        is omitted from the plot.
+
+        The `GetVisuals1D` class adds new visuals to a pre-existing `Visuals1D` object that is passed to its `__init__`
+        method. This only adds a new entry if the visual are not already in this object.
+
+        Parameters
+        ----------
+        include
+            Sets which 1D visuals are included on the figure that is to be plotted (only entries which are `True`
+            are extracted via the `GetVisuals1D` object).
+        visuals
+            The pre-existing visuals of the plotter which new visuals are added too via the `GetVisuals1D` class.
+        """
+        super().__init__(include=include, visuals=visuals)
+
+    def via_light_obj_from(self, light_obj: Union[LightProfile, Galaxy]) -> Visuals1D:
+        """
+        From an object with light profiles (e.g. a `LightProfile`, `Galaxy`) get its attributes that can be plotted 
+        and return them  in a `Visuals1D` object.
+
+        Only attributes not already in `self.visuals` and with `True` entries in the `Include1D` object are extracted
+        for plotting.
+
+        From a light object the following 1D attributes can be extracted for plotting:
+
+        - half_light_radius: the radius containing 50% of the light objects total integrated luminosity.
+
+        Parameters
+        ----------
+        light_obj
+            The light object (e.g. a `LightProfile`, `Galaxy`) whose attributes are extracted for plotting.
+
+        Returns
+        -------
+        Visuals1D
+            The collection of attributes that can be plotted by a `Plotter` object.
+        """
+
+        if light_obj is None:

```

Comment:
If a function is checking one of the arguments is None and returning a default value is might make more sense to check the argument before calling the function as this is more explicit. Obviously within reason to reduce duplication